### ISSUE TITLE
Regression in analyzer_benchmarksmoke_catalina_start_up

### ISSUE THREAD

**Issue by kf6gpe** [flutter/flutter#68716]
https://flutter-flutter-perf.skia.org/e/?begin=1603145146&end=1603212632&keys=Xc2dd62039544542694cd4f4e2a314b62&num_commits=50&request_type=1&xbaroffset=22098

Pretty clear signal at https://github.com/flutter/flutter/pull/68554.

Subresult `flutter_repo_batchflutter_repo_batch_minimumflutter_repo_watchflutter_repo_watch_maximumflutter_repo_watch_minimummega_gallery_watchmega_gallery_watch_maximumtimeToFirstFrameMicrostimeToFirstFrameRasterizedMicros`.

**Comment by kf6gpe** [comment]
Could this be related to https://github.com/flutter/flutter/issues/68718 as well? And really just a test change and not the engine roll?

**Comment by jonahwilliams** [comment]
There were several dart rolls in that engine roll, this seems real.

**Comment by jonahwilliams** [comment]
@devoncarew

**Comment by scheglov** [comment]
Most probably this is because of my changes.
I will revert them in https://dart-review.googlesource.com/c/sdk/+/168881

**Comment by jason-simmons** [comment]
Is the affected benchmark an analyzer benchmark or the smoke_catalina_start_up benchmark?  The dashboard link shows graphs of timeToFirstFrameMicros and timeToFirstFrameRasterizedMicros for smoke_catalina_start_up.

I can reproduce a change in the smoke_catalina_start_up metrics starting at the sound null safety change in the engine (https://github.com/flutter/engine/commit/5bd7260a1e778ba60b342f780049aeabddd45a7c)

On my Moto G4 setup, timeToFirstFrameMicros and timeToFirstFrameRasterizedMicros increased by around 10% with that commit.

@chinmaygarde

**Comment by scheglov** [comment]
My changes could only affect analyzer benchmarks.

**Comment by jason-simmons** [comment]
This might be related to the removal of the async isolate startup mechanism introduced in https://github.com/flutter/engine/commit/190fd8eb5131c76fdbda260846416e05aab669b9.

The smoke_catalina_start_up metrics dropped when that landed (see https://flutter-flutter-perf.skia.org/e/?begin=1602527014&end=1603333553&keys=Xc2dd62039544542694cd4f4e2a314b62&requestType=0&xbaroffset=22098)

However, in that graph the current benchmark values (with sound null safety) are higher than the ones from before async isolate startup.

**Comment by chinmaygarde** [comment]
> On my Moto G4 setup, timeToFirstFrameMicros and timeToFirstFrameRasterizedMicros increased by around 10% with that commit.

This may be because we now have to wait for the application kernel snapshots to be available before isolate launch can begin. Before, we would launch the isolate and later associate the application kernel files with the isolate before running user code in the isolate.

> This might be related to the removal of the async isolate startup mechanism...

Isolate startup is now still asynchronous, but starts after the application kernel files are available.

**Comment by zanderso** [comment]
We can't back out the sound null-safety change, so this regression will have to be addressed (if possible) by improvements to start up time going forward. Therefore, I'm going to reduce the priority of this issue to P3, but it could also be de-duped against an existing issue about improving start up performance if we have one.

**Comment by jason-simmons** [comment]
I spent some more time looking at this.

The `timeToFirstFrameMicros` metric is measuring the interval between the `FlutterEngineMainEnter` event in the `DartVM` constructor and the `Widgets built first useful frame` event in the framework `WidgetsBinding`.

The graph from the perf dashboard is showing a 40% increase in `timeToFirstFrameMicros`.  I haven't been able to reproduce that on the various devices I've tried.

The difference I'm seeing is 10-20%, and as far as I can tell it is attributable to the shift in when the root isolate is created.
